### PR TITLE
Fix join communities design review bugs

### LINKED ISSUE #16862: Design review: join community screen
- [x] 1. Small font - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510722370)  
- [x] 2. Additional line inside community logo **(Android only)** - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#511298876)  
- [x] 3. Top margin with opened bottom sheet - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510729314)  
- [x] 4. Wrong font size - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510728917) 
- [x] 5. Wrong font - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510728659)  
- [x] 6. The color inside the checkbox is lighter than in design - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510726719)

### PR DESCRIPTION
**Pull request by jo-mut**
fixes #16862

### CODE REVIEW

File: src/quo2/components/selectors/selectors/style.cljs [hunk 1]
```diff
@@ -24,7 +24,7 @@
 (defn checkbox-background-color
   [customization-color]
   {:normal {:checked   (colors/custom-color-by-theme customization-color 50 60)
-            :unchecked (colors/theme-colors colors/white-opa-40 colors/neutral-80-opa-40)}
+            :unchecked (colors/theme-colors colors/white-opa-30 colors/neutral-80-opa-40)}
```
Comment (Parveshdhull) [inline src/quo2/components/selectors/selectors/style.cljs]: Hi @jo-mut, Thank you very much for PR.
I am not sure if am looking into right file, but I don't see white-opa-30 in [checkbox figma compoent](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=398-18229&mode=design&t=X8QmqSRi4YZXqkRB-0)
But I do see white-opa-40 for disabled, unchecked variant.



if these changes refer to some other file, please link it

Comment (jo-mut) [reply src/quo2/components/selectors/selectors/style.cljs]: @Parveshdhull actually I will have to revert this change. The figma design i was referencing has been updating and this change is no longer valid. Its possible that I missed the correct color value. Thanks for confirming

File: src/status_im2/common/scroll_page/style.cljs [hunk 2]
```diff
@@ -68,6 +68,9 @@
...
 (def display-picture
-  {:border-radius picture-diameter
-   :width         picture-diameter
-   :height        picture-diameter})
+  {:border-radius    picture-diameter
+   :width            picture-diameter
+   :height           picture-diameter
+   :background-color (colors/theme-colors
+                      colors/white
+                      colors/neutral-95)})
```
Comment (ulisesmac) [inline src/status_im2/common/scroll_page/style.cljs]: These params fit in the same line :+1:

**Review by ulisesmac — APPROVED** [review]

File: src/status_im2/common/scroll_page/style.cljs [hunk 3]
```diff
@@ -68,6 +68,7 @@
     :left          (- (/ picture-radius 2) picture-border-width)}))
 
 (def display-picture
-  {:border-radius picture-diameter
-   :width         picture-diameter
-   :height        picture-diameter})
+  {:border-radius    picture-diameter
+   :width            picture-diameter
+   :height           picture-diameter
+   :background-color (colors/theme-colors colors/white colors/neutral-95)})
```
Comment (J-Son89) [inline src/status_im2/common/scroll_page/style.cljs]: can you pass `theme` as the 3rd arg here? 🙏

**Review by J-Son89 — APPROVED** [review]
left a small comment, other than that looks good 👍

**Review by Parveshdhull — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]

**Comment by pavloburykh** [comment]
@jo-mut Based on points described in [issue](https://github.com/status-im/status-mobile/issues/16862#issue-1834790029)

>1. Small font - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510722370)
>2. Additional line inside community logo (Android only) - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#511298876)
>3. Top margin with opened bottom sheet - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510729314)
>4. Wrong font size - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510728917)
>5. Wrong font - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510728659)
>6. The color inside the checkbox is lighter than in design - comment [here](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#510726719)

1 - do not see any changes compared to nightly build
2 - I have not reproduced it before so anything has changed here for me
3. - seems to be fixed
4. - Nothing changed, margin remained the same
5. - seem to be fixed
6 - do not see any changes compared to nightly build

I think it makes sense to ask @Francesca-G to review.

**Review by Francesca-G — COMMENTED** [review]
here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=5151%3A64611&mode=design&t=Z5pkcZzIYY6i6N18-1) with the review :)

**Comment by jo-mut** [comment]
@pavloburykh 
4 and 6 is okay now with in the latest commit. I may have missed it while merging conflicts. 
2 is only visible if the app is on light mode. It hard to notice on dark theme because of the its pretty much the same color

**Comment by jo-mut** [comment]
@pavloburykh @Francesca-G 
I did this overlay and it seems to me the fonts are well implemented. On the figma the font size is 13 which is rightly implemented on code

Screenshot 2023-08-31 at 20 47 46



the top view is iOS
the middle is the figma design
the bottom is android

it seems to me ISSUE 1 is a difference of how android calculates fonts but if you compare the iOS implementation with  android then it all okay

**Comment by Francesca-G** [comment]
@Francesca-G  sorry i edited your comment instead of replying to it. For visibility I have pasted your comment 

You're right, it looks like the size is correct but there might be some problem either with the weight or the kerning of the text. As you can see by overlapping it it looks perfect at first but then the word ends up being shorter than design, I believe that's the reason it appears smaller
264948848-eb50781c-4f6a-4e4f-aa63-eef709bc8f9a

**Comment by jo-mut** [comment]
@Francesca-G 
You are right but am afraid that this is something that affects the whole app not just this particular text. This is Figma uses its own font rendering because each device renders fonts differently depending on the device OS. In this case comparing the design and iOS implementation will be fairer since iOS is the default reference OS for pixel perfect implementation.

**Comment by jo-mut** [comment]
With iOS the look is seamless

Screenshot 2023-09-01 at 12 04 08

**Comment by churik** [comment]
@Francesca-G is it better in this PR than on nightly? 

If you can confirm that, we may merge PR and fix the remaining issues as follow-ups

**Comment by Francesca-G** [comment]
> @Francesca-G
> You are right but am afraid that this is something that affects the whole app not just this particular text. This is Figma uses its own font rendering because each device renders fonts differently depending on the device OS. In this case comparing the design and iOS implementation will be fairer since iOS is the default reference OS for pixel perfect implementation.

I'm comparing it to iOS in the screenshot I posted

**Comment by Francesca-G** [comment]
> @Francesca-G is it better in this PR than on nightly?
> 
> If you can confirm that, we may merge PR and fix the remaining issues as follow-ups

Yes, it's an improvement compared to nightly even though it's not pixel perfect

**Comment by jo-mut** [comment]
@pavloburykh  I suppose this pr then is ready for merge

**Comment by pavloburykh** [comment]
> @pavloburykh I suppose this pr then is ready for merge

if @Francesca-G approves - then it is ready for merge

**Review by Francesca-G — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hey @jo-mut! 

After merging of this PR it turned out that there are couple of changes that were not expected by this PR:

1. "Join community" button is replaced by "Request to join"

As far as I see there is a separate [PR](https://github.com/status-im/status-mobile/pull/17168) on that which is review column. So I wonder how these changes leaked in this PR?

photo_2023-09-06 11 20 03
photo_2023-09-06 11 20 07

2. Checkbox is removed from Community rules bottom sheet

I don't see any mention in this PR about this change (correct me I am wrong). 

photo_2023-09-06 11 20 10

When I tested the build from August 31 there were none of these changes. Also I don't see any requests from @Francesca-G regarding these changes in this PR. Am I missing something?

One of the problems is that such unexpected changes which are silently merged into develop without manual qa approval (or at least running of e2e) can brake e2e tests or can potentially cause some other bugs. 

I see that community join flow has been updated in [Figma](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=14459-153488&mode=design&t=xN63vJuNYJ9D1I4f-0) But we should implement any updates according to our standard process: creating an issue with description, creating corresponding PR, reviewing the PR, passing e2e test, approving from QA team (if needed), passing design review.

Maybe I have missed some conversations/agreements regarding those changes. Will appreciate if you clarify on that. Thank you!

**Comment by jo-mut** [comment]
@pavloburykh 
https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=650%3A30484&mode=design&t=xYvZnFe4nPlaTRkn-1

The figma link above removed the checkbox from the community rules and the feeback figmas were updated like this without the checkbox
Screenshot 2023-09-06 at 11 47 19

**Comment by jo-mut** [comment]
As for the changes on join community button there is an issue for that and a pr. https://github.com/status-im/status-mobile/issues/17065. I may have leaked some of those changes into this pr which is my mistake

**Comment by churik** [comment]
it contradicts with another PR which was merged one week ago. 
https://github.com/status-im/status-mobile/pull/17091 - which also changed flow for joining communities

cc @cammellos

**Comment by jo-mut** [comment]
@churik  with the updated figma there is no more `Join Community` since what really happen is the user always sends a request to join which maybe auto-accepted or manually accepted. This is why the figma were updated. Am not sure where the contradictions

**Comment by churik** [comment]
@jo-mut anyway if drastic changes like this are made after design review, please, re-ask for manual QA, as currently we have all community e2e broken in develop.

Thank you.

**Comment by jo-mut** [comment]
thank you

**Comment by pavloburykh** [comment]
Hey @jo-mut! Also, I see "Follow up" label in this PR. If you have already created followup please link it to this PR. If not - please create one according to our process. Thank you.

**Comment by jo-mut** [comment]
thats right. thanks

**Pull request merged**